### PR TITLE
algorithm: Liouville lambda function

### PR DESCRIPTION
**Pull request by itsAkshayDubey**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/liouville_lambda.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if not isinstance(number, int):
+        raise TypeError(f"Input value of [number={number}] must be an integer")
+    if number < 0:
+        raise ValueError("Input must be positive")
+    if number == 0:
+        raise ValueError("Input must be non-zero")
+    if len(prime_factors(number)) % 2 == 0:
+        return 1
+    else:
+        return -1
```
Comment (cclauss) [inline maths/liouville_lambda.py:45]:
```suggestion
    return -1 if len(prime_factors(number)) % 2 else 1
```

File: maths/liouville_lambda.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    Traceback (most recent call last):
+        ...
+    TypeError: Input value of [number=11.0] must be an integer
+    """
+    if not isinstance(number, int):
+        raise TypeError(f"Input value of [number={number}] must be an integer")
+    if number < 0:
+        raise ValueError("Input must be positive")
+    if number == 0:
+        raise ValueError("Input must be non-zero")
```
Comment (cclauss) [inline maths/liouville_lambda.py:41]:
```suggestion
    if number < 1:
        raise ValueError("Input must be a positive integer")
```

File: maths/liouville_lambda.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+https://en.wikipedia.org/wiki/Liouville_function
+"""
+
+# Author : Akshay Dubey (https://github.com/itsAkshayDubey)
+from maths.prime_factors import prime_factors
+
+
+def liouville_lambda(number: int) -> int:
+    """
+    # doctest: +NORMALIZE_WHITESPACE
```
Comment (CaedenPH) [inline maths/liouville_lambda.py:16]: This isn't necessary here
```suggestion
```

**Review by CaedenPH — COMMENTED** [review]
Small change, otherwise looks good

File: maths/liouville_lambda.py (hunk 1 above)
Comment (itsAkshayDubey) [reply maths/liouville_lambda.py]: Fixed in [this](https://github.com/TheAlgorithms/Python/pull/7986/commits/f9df30f79a59beb2fdc269ad78595d3bd9e029b7) commit. Kindly have a look.

Thanks

File: maths/liouville_lambda.py (hunk 2 above)
Comment (itsAkshayDubey) [reply maths/liouville_lambda.py]: Fixed in [this](https://github.com/TheAlgorithms/Python/pull/7986/commits/d594f79d3908c9baca0c57b58ade20badc935689) commit. Kindly have a look.

Thanks

File: maths/liouville_lambda.py (hunk 3 above)
Comment (itsAkshayDubey) [reply maths/liouville_lambda.py]: Fixed in [this](https://github.com/TheAlgorithms/Python/pull/7986/commits/8e110710d29e7d88478f349cbdfb96f3ef8c5870) commit. Kindly have a look.

Thanks

**Review by cclauss — APPROVED** [review]

**Pull request merged**